syntax = "proto3";

package dresscast.v1;

message Coordinate {
  double lat = 1;
  double lon = 2;
}

message WeatherData {
  double temperature = 1;
  double humidity = 2;
  double wind_speed = 3;
  double precipitation = 4;
  string time = 5;
}

message City {
  int32 id = 1;
  int32 user_id = 2;
  string name = 3;
  double lat = 4;
  double lon = 5;
}

message User {
  int32 id = 1;
  string username = 2;
  optional string telegram_id = 3;
  bool is_admin = 4;
  string locale = 5;
}
